### PR TITLE
Strip logs at compile time

### PR DESCRIPTION
**Pull request by averne**
This strips logs which level is lower than `LV_LOG_LEVEL` at compile time.
Besides reducing the executable size, it removes some unnecessary function calls.

### CODE REVIEW

**Comment by embeddedt** [comment]
(cc @kisvegabor)

Not sure if this is worth the additional macro complexity.

File: lv_misc/lv_log.h [hunk 1]
```diff
@@ -64,10 +64,26 @@ void lv_log_add(lv_log_level_t level, const char * file, int line, const char *
...
  **********************/
 
-#define LV_LOG_TRACE(dsc)   lv_log_add(LV_LOG_LEVEL_TRACE, __FILE__, __LINE__, dsc);
-#define LV_LOG_INFO(dsc)    lv_log_add(LV_LOG_LEVEL_INFO, __FILE__, __LINE__, dsc);
-#define LV_LOG_WARN(dsc)    lv_log_add(LV_LOG_LEVEL_WARN, __FILE__, __LINE__, dsc);
-#define LV_LOG_ERROR(dsc)   lv_log_add(LV_LOG_LEVEL_ERROR, __FILE__, __LINE__, dsc);
+#if LV_LOG_LEVEL <= LV_LOG_LEVEL_TRACE
+#  define LV_LOG_TRACE(dsc)   lv_log_add(LV_LOG_LEVEL_TRACE, __FILE__, __LINE__, dsc);
+#else
+#  define LV_LOG_TRACE(dsc)   {;}
```
Comment (embeddedt) [inline lv_misc/lv_log.h:70]: Usually I use `do {} while(0)` here; are you sure this has the same effect?

**Comment by BrianPugh** [comment]
This is pretty standard of logging macros; I'm in favor for it.

File: lv_misc/lv_log.h (hunk 1 above)
Comment (averne) [reply lv_misc/lv_log.h:70]: This is used later in the same file.

Comment (embeddedt) [reply lv_misc/lv_log.h:70]: @averne Fair enough.

**Comment by kisvegabor** [comment]
I also like the proposed update. Now `LV_LOG_...` macros hides `lv_log_add` and the log texts are passed to the function regardless to the log level. Therefore the compiler has no chance the drop the log messages below the current log level.

So IMO it's worth this little extra complexity.

**Comment by kisvegabor** [comment]
@embeddedt So I like this update. Is there any objection to merge it?

**Comment by embeddedt** [comment]
No, but I just realized it is aimed for `master` not `dev-6.0`.

**Comment by kisvegabor** [comment]
@embeddedt 
> it is aimed for master not dev-6.0.

Thank you for pointing that out!


@averne After changing the base to `dev-6.0` a lot of conflicts occurred. So I manually added your updates. See: b71b1f1

Thank you very much!

**Comment by averne** [comment]
No problem at all. Thanks for all the work you put in this great lib!

**Pull request closed without merging**